File: CS/AuditDemo.Module/WebServiceTimestampStrategy.cs
using System;
using DevExpress.Persistent.AuditTrail;
using DevExpress.Xpo;
using AuditDemo.Module.TimestampWebService;

namespace AuditDemo.Module {
	public class WebServiceTimestampStrategy : IAuditTimestampStrategy {
		DateTime cachedTimeStamp;
		#region IAuditTimeStampStrategy Members
		public DateTime GetTimestamp(AuditDataItem auditDataItem) {
			return cachedTimeStamp;
		}
		public void OnBeginSaveTransaction(Session session) {
			try {
				TimestampService service = new TimestampService();
				cachedTimeStamp = service.GetTime();
			}
			catch {
                throw new Exception("Cannot access the TimeStampWebService. Make sure it's running.");
			}
		}
		#endregion
	}
}

File: VB/AuditDemo.Module/WebServiceTimestampStrategy.vb
Imports System
Imports DevExpress.Persistent.AuditTrail
Imports DevExpress.Xpo
Imports AuditDemo.Module.TimestampWebService

Namespace AuditDemo.Module
	Public Class WebServiceTimestampStrategy
		Implements IAuditTimestampStrategy

		Private cachedTimeStamp As DateTime
		#Region "IAuditTimeStampStrategy Members"
		Public Function GetTimestamp(ByVal auditDataItem As AuditDataItem) As DateTime
			Return cachedTimeStamp
		End Function
		Public Sub OnBeginSaveTransaction(ByVal session As Session)
			Try
				Dim service As New TimestampService()
				cachedTimeStamp = service.GetTime()
			Catch
				Throw New Exception("Cannot access the TimeStampWebService. Make sure it's running.")
			End Try
		End Sub
		#End Region
	End Class
End Namespace
